### PR TITLE
chore: update grpc queries to handle multiple fees

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Addresses updating of grpc queries and tests following model changes

closes: #953 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by colin-axner — COMMENTED** [review]
LGTM! Will approve once upstream dependencies are merged.

**Review by seantking — APPROVED** [review]

File: proto/ibc/applications/fee/v1/query.proto
Comment (AdityaSripal) [inline proto/ibc/applications/fee/v1/query.proto:54]: This query doesn't make much sense to me. Why include packet id in response when it is already in request?

Makes more sense to have a PacketFee RPC that takes packetID and returns the packet fees for it.

Also I think it makes more sense to have an RPC that just aggregates all packet fees for a given packetID.

Something like

`QueryRecvFee(packetID)` => `total receive fees earned for this packet`

and similar for ack and timeout

**Review by AdityaSripal — APPROVED** [review]
Looks good, but I think we should add other queries that will be more useful in future prs

File: proto/ibc/applications/fee/v1/query.proto
Comment (damiannolan) [reply proto/ibc/applications/fee/v1/query.proto:54]: I totally agree, my main goal with this PR was to just fix what was there w.r.t. model changes that were made. 
I think we should definitely follow up on fleshing out the query api. I really like the idea of:
`QueryRecvFee(packetID)` => `total receive fees earned for this packet`
`QueryAckFee(packetID)` ...
`QueryTimeoutFee(packetID)` ...

If you want I can update this response type to include just `PacketFees` (the list of packet fees without the packet ID) instead of the identified type.

File: modules/apps/29-fee/keeper/grpc_query_test.go
Comment (colin-axner) [inline modules/apps/29-fee/keeper/grpc_query_test.go:124]: haha I just thought to do the same on #983

File: modules/apps/29-fee/keeper/grpc_query.go
Comment (colin-axner) [inline modules/apps/29-fee/keeper/grpc_query.go:39]: Is this correct? From my investigation, pagination doesn't include the prefix iterated on in the return value. I guess there's a trailing "/" leading the key, which is subtle

So I think the key is something like "/transfer/channel-0/5"

**Review by colin-axner — APPROVED** [review]

File: modules/apps/29-fee/keeper/grpc_query.go
Comment (colin-axner) [reply modules/apps/29-fee/keeper/grpc_query.go:39]: I'm kinda wondering if we should just reconstruct the full key and then use the parse functions in keys.go

Comment (damiannolan) [reply modules/apps/29-fee/keeper/grpc_query.go:39]: Yeah, correct, so a paginated query doesn't include the prefix, but because of the leading `"/"` when we split using the slash separator the first element is an empty string. So with your example, `keySplit` ends up as a 4 element slice: 
`[]string{"", "transfer", "channel-0", "5"}`.

I do think since you've added the parse function we should try to leverage it here. It should be something like below, right?
```
packetID, err := types.ParseKeyFeesInEscrow(types.FeesInEscrowPrefix + string(key))
```

**Pull request merged**
